### ISSUE TITLE
Setup gets stuck and doesn't proceed further : Tine 2018.08.3 (All in one zip package)

### ISSUE THREAD

**Issue by jameswill** [tine20/tine20#6906]
Hi,

I am trying to manually install Tine 2018.08.3 (all in one package) on : 
OS : Centos : 7.1
WebServer : Apache 2.2.32
PHP :5.6.31
MYSQL : 5.6.39

However i face an issue while doing so . After entering the configuration details in config.inc.php
and hitting : 
http://url-to-tine/setup.php

The setup keeps on loading the page and doesn't proceed any further .

![test](https://user-images.githubusercontent.com/6368036/44399873-c0852a80-a566-11e8-88a7-e6333890f645.png)

What could be causing this issue . I checked on other servers as well .Still getting the same issue.

Could you please look into this.?.

**Comment by lab-at-nohl** [comment]
Hi, we need some more information. Otherwise it's only a lucky shot. 

1. Have you activate the log in config.inc.php, level DEBUG? 
```php
  'logger' =>
  array (
#    'active' => false,
    'active' => true,
    'filename' => '/your writable path/',
    'priority' => 7, # 9=TRACE, 7=DEBUG, 5=NOTICE, 3=ERR
    'filter' => array (),
```
2. Is there any hint in javascript console of your browser? Are you using SSL? Atm we have a problem to load JS-files (try all in the source), if so try in your Apache 2.x VirtualHost for SSL to add:
```apache
 SetEnv HTTPS on
```

Any more information?

**Comment by jameswill** [comment]
Hi @lab-at-nohl ,

Thanks for the information regarding the debug. No i am not using SSL. As per your advise i enabled debug in config.inc.php and this is what the log generated in the log file . There is no hint  :


45a11 -- none -- - 2018-08-22T03:55:34+00:00 INFO (6): Tinebase_Core::setupLogger::797 Logger initialized.
45a11 -- none -- - 2018-08-22T03:55:34+00:00 INFO (6): index.php (20) Start processing request (PID: 13626)
45a11 -- none -- - 2018-08-22T03:55:34+00:00 INFO (6): Tinebase_Core::setupCache::926 Cache disabled
45a11 -- none -- - 2018-08-22T03:55:34+00:00 DEBUG (7): Tinebase_Core::setExecutionLifeTime::1318 setting execution life time to: 0
45a11 -- none -- - 2018-08-22T03:55:34+00:00 DEBUG (7): Tinebase_Core::createAndConfigureDbAdapter::1039 Creating Pdo_Mysql DB adapter (db name: {{TINEDB}})
45a11 -- none -- - 2018-08-22T03:55:34+00:00 DEBUG (7): Tinebase_Core::createAndConfigureDbAdapter::1057 Using MySQL charset: utf8mb4
45a11 -- none -- - 2018-08-22T03:55:34+00:00 DEBUG (7): Tinebase_Core::createAndConfigureDbAdapter::1075 Falling back to utf-8 charset
45a11 -- none -- - 2018-08-22T03:55:34+00:00 INFO (6): Tinebase_Core::initFramework::456 initializing framework (PID: 13626)
45a11 -- none -- - 2018-08-22T03:55:34+00:00 WARN (4): Tinebase_Core::setupStreamWrapper::968 Filesdir config value not set. tine20:// streamwrapper not registered, virtual filesystem not available.
45a11 -- none -- - 2018-08-22T03:55:34+00:00 DEBUG (7): Tinebase_Core::setupUserLocale::1200 given localeString 'auto'
45a11 -- none -- - 2018-08-22T03:55:34+00:00 DEBUG (7): Tinebase_Core::setupUserLocale::1227 Try to detect the locale of the user (browser, environment, default)
45a11 -- none -- - 2018-08-22T03:55:34+00:00 DEBUG (7): Tinebase_Translation::getLocale::199 given localeString 'auto'
45a11 -- none -- - 2018-08-22T03:55:34+00:00 DEBUG (7): Tinebase_Translation::getLocale::212 'en_US' is not supported, checking fallback
45a11 -- none -- - 2018-08-22T03:55:34+00:00 DEBUG (7): Tinebase_Translation::getLocale::238 selected locale: 'en'
45a11 -- none -- - 2018-08-22T03:55:34+00:00 INFO (6): Tinebase_Core::setupUserLocale::1251 Setting user locale: en
45a11 -- none -- - 2018-08-22T03:55:34+00:00 DEBUG (7): Tinebase_Core::setupUserLocale::1258 Setting CTYPE locale from "C" to "en_US.UTF8".
45a11 -- none -- - 2018-08-22T03:55:34+00:00 ERR (3): Tinebase_Exception::logExceptionToLogger::133 Zend_Db_Statement_Exception -> SQLSTATE[42S02]: Base table or view not found: 1146 Table '{{TINEDB}}.{{DBPREFIX}}applications' doesn't exist, query was: SELECT `applications`.* FROM `{{DBPREFIX}}applications` AS `applications`
45a11 -- none -- - 2018-08-22T03:55:34+00:00 ERR (3): Tinebase_Exception::logExceptionToLogger::155 #0 .../vendor/zendframework/zendframework1/library/Zend/Db/Statement.php(303): Zend_Db_Statement_Pdo->_execute(Array)
#1 .../vendor/zendframework/zendframework1/library/Zend/Db/Adapter/Abstract.php(480): Zend_Db_Statement->execute(Array)
#2 .../vendor/zendframework/zendframework1/library/Zend/Db/Adapter/Pdo/Abstract.php(238): Zend_Db_Adapter_Abstract->query(Object(Zend_Db_Select), Array)
#3 .../Tinebase/Backend/Sql/Abstract.php(795): Zend_Db_Adapter_Pdo_Abstract->query(Object(Zend_Db_Select))
#4 .../Tinebase/Backend/Sql/Abstract.php(551): Tinebase_Backend_Sql_Abstract->_fetch(Object(Zend_Db_Select), 'fetch_all')
#5 .../Tinebase/Application.php(206): Tinebase_Backend_Sql_Abstract->search(NULL, NULL)
#6 .../Tinebase/Core.php(531): Tinebase_Application->getApplications()
#7 .../Tinebase/Core.php(507): Tinebase_Core::getPreCompiledContainer()
#8 .../Tinebase/Core.php(489): Tinebase_Core::setupContainer()
#9 .../Tinebase/Server/Http.php(58): Tinebase_Core::initFramework()
#10 .../Tinebase/Core.php(261): Tinebase_Server_Http->handle(Object(Tinebase_Http_Request))
#11 .../index.php(24): Tinebase_Core::dispatchRequest()
#12 {main}
45a11 -- none -- - 2018-08-22T03:55:34+00:00 NOTICE (5): Setup_Controller::setupRequired::906 SQLSTATE[42S02]: Base table or view not found: 1146 Table '{{TINEDB}}.{{DBPREFIX}}applications' doesn't exist, query was: DESCRIBE `{{DBPREFIX}}applications`
45a11 -- none -- - 2018-08-22T03:55:34+00:00 DEBUG (7): Tinebase_Server_Http::handle::121 Setup required
45a11 -- none -- - 2018-08-22T03:55:34+00:00 DEBUG (7): Tinebase_Frontend_Http::Tinebase_Frontend_Http::setupRequired (230) Update/Setup required!

Regards.

**Comment by pschuele** [comment]
this looks a bit odd: {{TINEDB}}.{{DBPREFIX}}applications

when exactly does the loading animation show? after entering the setup login credentials or before? do you see any errors in the webserver logs (for example apache/nginx error.log)?

**Comment by jameswill** [comment]
@pschuele 
 
i have entered the db details ,username and password in config.inc.php file .

The loading animation comes after i hit  the setup url http://url-to-tine/setup.php . Tried creating blank config.inc.php file as well and giving permissions to that file .
Even that doesn't seem to work .
Also  there is no error generated in apache log .

**Comment by pschuele** [comment]
which browser are you using?

**Comment by jameswill** [comment]
For testing i used both the browsers i.e Chrome and Firefox.

**Comment by pschuele** [comment]
hm, I'm running out of ideas here :(

maybe you could try to install tine20 via the command line:

```bash
php setup.php --config /path/to/config/file --install
```

and see what happens...

**Comment by lab-at-nohl** [comment]
As this seems to be an DB problem, can you elaborate on this? Is "_{{TINEDB}}.{{DBPREFIX}}_" and similar a substitution that you made? 

> 45a11 -- none -- - 2018-08-22T03:55:34+00:00 DEBUG (7): Tinebase_Core::createAndConfigureDbAdapter::1039 Creating Pdo_Mysql DB adapter (db name: {{TINEDB}})

Can you double check your DB settings and post it here? Can you try to install in a new, clean DB (with no tables) then? I ask myself how the installer comes to this point when it should instead create the table structures (it is your first try, not an update - right?)...

I am working in a similiar config. But CentOS 7.5 atm and PHP 7.0. Aren't you using MariaDB? Please reinstall all files from the zip, as you're on CentOS make sre SELINUX isn't messing with your permissions. Also make sure that PHP extensions like PDO are installed and available to your php (-fpm?). 

If you have SSH try pschuele's advice first...

**Comment by jameswill** [comment]
Hi,

I am facing same issue in Tine 2018084 as well.

However I tried installing with command line and it worked well.
But Ideally it should work with browser based installation also.

**Comment by gohrner** [comment]
I seem to have an issue with identical symptoms, but possibly different cause. I installed the Debian 9 packages (successfully after leaving spaces out of my password...), but I also only get the spinning ring when invoking setup.php.

The tine20 DB has been created by the setup scripts, but I get the following in the logs:

`4a4ec -- none -- - 2018-09-02T12:40:06+00:00 ERR (3): Tinebase_Exception::logExceptionToLogger::133 Zend_Db_Statement_Exception -> SQLSTATE[42S02]: Base table or view not found: 1146 Table 'tine20.tine20_applications' doesn't exist, query was: SELECT `applications`.* FROM `tine20_applications` AS `applications`
4a4ec -- none -- - 2018-09-02T12:40:06+00:00 ERR (3): Tinebase_Exception::logExceptionToLogger::155 #0 .../vendor/zendframework/zendframework1/library/Zend/Db/Statement.php(303): Zend_Db_Statement_Pdo->_execute(Array)
#1 .../vendor/zendframework/zendframework1/library/Zend/Db/Adapter/Abstract.php(480): Zend_Db_Statement->execute(Array)
#2 .../vendor/zendframework/zendframework1/library/Zend/Db/Adapter/Pdo/Abstract.php(238): Zend_Db_Adapter_Abstract->query('SELECT `applica...', Array)
#3 .../Tinebase/Backend/Sql/Abstract.php(797): Zend_Db_Adapter_Pdo_Abstract->query(Object(Zend_Db_Select))
#4 .../Tinebase/Backend/Sql/Abstract.php(553): Tinebase_Backend_Sql_Abstract->_fetch(Object(Zend_Db_Select), 'fetch_all')
#5 .../Tinebase/Application.php(206): Tinebase_Backend_Sql_Abstract->search(NULL, NULL)
#6 .../Tinebase/Core.php(531): Tinebase_Application->getApplications()`

I would have expected setup.php to create all required tables if the Debian Package installation does not do so...

**Comment by gohrner** [comment]
Command line installation also seems to have worked for me, all tables are created now and the web UI is accessible.

**Comment by pschuele** [comment]
ah, it might be an issue of missing language information. see https://github.com/tine20/tine20/commit/4553f7518681753ea13fea440937b9423fe5d0a9

could you try to apply the change (patch) on your system and try again?

**Comment by jameswill** [comment]
Hi,

I tried adding given patch in Http.php file. However issue still persists.

**Comment by pschuele** [comment]
> I tried adding given patch in Http.php file. However issue still persists.

ok, I'll try to reproduce this.

**Comment by pschuele** [comment]
i can't reproduce the problem. just did a fresh install of the tine20 ubuntu package on 18.04 and I could log in to setup.php without any problem.

![screenshot from 2018-09-06 16-47-45](https://user-images.githubusercontent.com/353947/45165362-f8da6900-b1f4-11e8-8f11-8618e93b3cec.png)

**Comment by jameswill** [comment]
HI,

We have downloaded all in one package from [http://packages.tine20.org/source/2018.08.4/tine20-allinone_2018.08.4.zip](url) and tried installing on linux server with centos 6,  php 5.6.x and mysql 5.6.x

**Comment by pschuele** [comment]
ok, maybe there is an issue with centos. I did not try that. will do when I find some time for that.
but centos 6 is kind of old. why not try with 7?

**Comment by SIRSteiner** [comment]
Same here at Hoster Strato AND Alfahosting with package all-in-one 2018.084

[tine.log](https://github.com/tine20/tine20/files/2428969/tine.log)

**Comment by lab-at-nohl** [comment]
Maybe this behavior is related to the latest changes regarding utf8-handling in DB (utf8mb4)... I have seen similar elsewhere. What is inside config.inc.php (except the password)?

**Comment by SIRSteiner** [comment]
config.inc.php (renamed to txt for upload):

[config.inc.php.txt](https://github.com/tine20/tine20/files/2430532/config.inc.php.txt)

MySQL-Server:
    Server: alfaxxxx.alfahosting-server.de (Localhost via UNIX socket)
    Server-Typ: MySQL
    Server-Version: 5.7.21 - MySQL Community Server (GPL)
    Protokoll-Version: 10
    Benutzer: webxxxxxxxx@localhost
    Server-Zeichensatz: UTF-8 Unicode (utf8)
utf8mb4_unicode_ci

**Comment by lab-at-nohl** [comment]
Ok, we can‘t compare directly. I‘m on a fully patched CentOS 7.5 using MariaDB 5.5.60-1.

My default collation is utf8_general_ci. Can you try that? I realize it is a wild guess (seeing your Firefox output), but I once had a problem regarding utf8 resulting in a „1146 ... table not found“-error.

**Comment by SIRSteiner** [comment]
Thank you but collation "utf8_general_ci" don't solve that problem.

**Comment by SIRSteiner** [comment]
I tried:
1. Update from 2018.08.2 to 2018.08.3 and to 2018.08.4 and it works
2. Setup 2018.08.4 via cli and it fails with the following error:

> Fatal error: Uncaught PDOException: SQLSTATE[HY000] [2002] Connection refused in /var/www/web22xxxxxx/html/beisteiners.de/tine/vendor/zendframework/zendframework1/library/Zend/Db/Adapter/Pdo/Abstract.php:128
> Stack trace:
#0 /var/www/web22xxxxxx/html/beisteiners.de/tine/vendor/zendframework/zendframework1/library/Zend/Db/Adapter/Pdo/Abstract.php(128): PDO->__construct('mysql:host=loca...', 'web22xxxxxx', 'password', Array)
>#1 /var/www/web22xxxxxx/html/beisteiners.de/tine/vendor/zendframework/zendframework1/library/Zend/Db/Adapter/Pdo/Mysql.php(111): Zend_Db_Adapter_Pdo_Abstract->_connect()
>#2 /var/www/web22xxxxxx/html/beisteiners.de/tine/Tinebase/Backend/Sql/Adapter/Pdo/Mysql.php(32): Zend_Db_Adapter_Pdo_Mysql->_connect()
>#3 /var/www/web22xxxxxx/html/beisteiners.de/tine/vendor/zendframework/zendframework1/library/Zend/Db/Adapter/Abstract.php(460): Tinebase_Backend_Sql_Adapter_Pdo_Mysql->_connect()
>#4 /var/www/web22xxxxxx/html/beisteiners.de/tine/vendor/zendframework/zendframework1/library/Zend/Db/Adapter/Pdo/Abstract.php(23 in /var/www/web22xxxxxx/html/beisteiners.de/tine/vendor/zendframework/zendframework1/library/Zend/Db/Adapter/Pdo/Abstract.php on line 144

I don't know if there is a relation between this error and the frontend problem?

**Comment by pschuele** [comment]
closing this issue. if you still have those problems in 2018.08.5, please reopen it and/or give us a comment.

**Comment by Smuggl** [comment]
Yesterday I tried a fresh install of the latest package (2018.08.05) and hit the same problem.
As per debug log the issue discussed above seems to persist. Please see attached debug.log file.
[debug.log](https://github.com/tine20/tine20/files/2556896/debug.log)

**Comment by pschuele** [comment]
i think, this is a duplicate of #6948 (which will be hopefully fixed in 2018.08.6 - coming next week).
so, please try again with the upcoming version and give us feedback if it works or not. thanks!
